diff --git a/ggml/src/ggml-cpu/arch/arm/repack.cpp b/ggml/src/ggml-cpu/arch/arm/repack.cpp
index a75344430..5f0d3843b 100644
--- a/ggml/src/ggml-cpu/arch/arm/repack.cpp
+++ b/ggml/src/ggml-cpu/arch/arm/repack.cpp
@@ -3749,13 +3749,14 @@ void ggml_gemm_q5_K_8x4_q8_K(int                        n,
     ggml_gemm_q5_K_8x4_q8_K_generic(n, s, bs, vx, vy, nr, nc);
 }
 
-void ggml_gemm_q4_K_8x8_q8_K(int                        n,
-                             float * GGML_RESTRICT      s,
-                             size_t                     bs,
-                             const void * GGML_RESTRICT vx,
-                             const void * GGML_RESTRICT vy,
-                             int                        nr,
-                             int                        nc) {
+#if defined(__aarch64__) && defined(__ARM_FEATURE_SVE) && defined(__ARM_FEATURE_MATMUL_INT8)
+static __attribute__((noinline)) void ggml_gemm_q4_K_8x8_q8_K_sve(int                        n,
+                                                                  float * GGML_RESTRICT      s,
+                                                                  size_t                     bs,
+                                                                  const void * GGML_RESTRICT vx,
+                                                                  const void * GGML_RESTRICT vy,
+                                                                  int                        nr,
+                                                                  int                        nc) {
     constexpr int qk = QK_K;
     const int     nb = n / qk;
 
@@ -4079,6 +4080,36 @@ void ggml_gemm_q4_K_8x8_q8_K(int                        n,
         return;
     }
 #endif  // SVE compile-time end
+}
+#endif
+
+void ggml_gemm_q4_K_8x8_q8_K(int                        n,
+                             float * GGML_RESTRICT      s,
+                             size_t                     bs,
+                             const void * GGML_RESTRICT vx,
+                             const void * GGML_RESTRICT vy,
+                             int                        nr,
+                             int                        nc) {
+    constexpr int qk = QK_K;
+    const int     nb = n / qk;
+
+    constexpr int ncols_interleaved = 8;
+    constexpr int blocklen          = 8;
+
+    assert(n % qk == 0);
+    assert(nr % 4 == 0);
+    assert(nc % ncols_interleaved == 0);
+
+    UNUSED(nb);
+    UNUSED(ncols_interleaved);
+    UNUSED(blocklen);
+
+#if defined(__aarch64__) && defined(__ARM_FEATURE_SVE) && defined(__ARM_FEATURE_MATMUL_INT8)
+    if (svcntb() * 8 == 256) {
+        ggml_gemm_q4_K_8x8_q8_K_sve(n, s, bs, vx, vy, nr, nc);
+        return;
+    }
+#endif
 
 #if defined(__aarch64__) && defined(__ARM_NEON) && defined(__ARM_FEATURE_MATMUL_INT8)
     constexpr int    q8_k_blocklen = 4;
