diff --git a/ggml/src/ggml-cpu/arch/arm/repack.cpp b/ggml/src/ggml-cpu/arch/arm/repack.cpp
index 30002f8..16ed27e 100644
--- a/ggml/src/ggml-cpu/arch/arm/repack.cpp
+++ b/ggml/src/ggml-cpu/arch/arm/repack.cpp
@@ -785,6 +785,25 @@ void ggml_gemv_q4_K_8x8_q8_K(int                        n,
                     uint8x16_t q4_qs_cp_2 = vld1q_u8(q4_base + 16 * cp + 128);
                     uint8x16_t q4_qs_cp_3 = vld1q_u8(q4_base + 16 * cp + 192);
 
+#if defined(__ARM_FEATURE_MATMUL_INT8)
+                    acc_lo[cp] = vusmmlaq_s32(
+                        acc_lo[cp], vandq_u8(q4_qs_cp_0, m4b), q8_qs[0]);
+                    acc_lo[cp] = vusmmlaq_s32(
+                        acc_lo[cp], vandq_u8(q4_qs_cp_1, m4b), q8_qs[1]);
+                    acc_lo[cp] = vusmmlaq_s32(
+                        acc_lo[cp], vandq_u8(q4_qs_cp_2, m4b), q8_qs[2]);
+                    acc_lo[cp] = vusmmlaq_s32(
+                        acc_lo[cp], vandq_u8(q4_qs_cp_3, m4b), q8_qs[3]);
+
+                    acc_hi[cp] = vusmmlaq_s32(
+                        acc_hi[cp], vshrq_n_u8(q4_qs_cp_0, 4), q8_qs[4]);
+                    acc_hi[cp] = vusmmlaq_s32(
+                        acc_hi[cp], vshrq_n_u8(q4_qs_cp_1, 4), q8_qs[5]);
+                    acc_hi[cp] = vusmmlaq_s32(
+                        acc_hi[cp], vshrq_n_u8(q4_qs_cp_2, 4), q8_qs[6]);
+                    acc_hi[cp] = vusmmlaq_s32(
+                        acc_hi[cp], vshrq_n_u8(q4_qs_cp_3, 4), q8_qs[7]);
+#else
                     acc_lo[cp] =
                         ggml_vdotq_s32(acc_lo[cp], vreinterpretq_s8_u8(vandq_u8(q4_qs_cp_0, m4b)), q8_qs[0]);  // 0 .. 7
                     acc_lo[cp] =
@@ -802,5 +821,6 @@ void ggml_gemv_q4_K_8x8_q8_K(int                        n,
                     acc_hi[cp] =
                         ggml_vdotq_s32(acc_hi[cp], vreinterpretq_s8_u8(vshrq_n_u8(q4_qs_cp_3, 4)), q8_qs[7]);  // 56..63
+#endif
                 }
 
                 // Iterates over a pair of column pairs (4 columns) to use a single 128 register
@@ -813,11 +833,19 @@ void ggml_gemv_q4_K_8x8_q8_K(int                        n,
 
                     // 0123 or 4567
                     float32x4_t sumf_0 =
+#if defined(__ARM_FEATURE_MATMUL_INT8)
+                        vcvtq_f32_s32(vmulq_s32(vmovl_s16(group_scales_lo), vuzp1q_s32(acc_lo[p], acc_lo[p + 1])));
+#else
                         vcvtq_f32_s32(vmulq_s32(vmovl_s16(group_scales_lo), vpaddq_s32(acc_lo[p], acc_lo[p + 1])));
+#endif
                     acc_f32[i] = vfmaq_f32(acc_f32[i], sb_scale, sumf_0);
 
                     float32x4_t sumf_1 =
+#if defined(__ARM_FEATURE_MATMUL_INT8)
+                        vcvtq_f32_s32(vmulq_s32(vmovl_s16(group_scales_hi), vuzp1q_s32(acc_hi[p], acc_hi[p + 1])));
+#else
                         vcvtq_f32_s32(vmulq_s32(vmovl_s16(group_scales_hi), vpaddq_s32(acc_hi[p], acc_hi[p + 1])));
+#endif
                     acc_f32[i] = vfmaq_f32(acc_f32[i], sb_scale, sumf_1);
                 }
 
